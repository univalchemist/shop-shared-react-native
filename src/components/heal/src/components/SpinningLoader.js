import React from 'react';
import { View, ViewPropTypes, Platform, ActivityIndicator } from 'react-native';
import Spinner from 'react-native-spinkit';
import theme from '@theme';

const SpinningLoader = ({ style }) => {
  if (Platform.OS === 'ios') {
    return (
      <View style={[styles.spinnerStyle, style]}>
        <Spinner
          isVisible={true}
          color={theme.heal.colors.crimson}
          size={36}
          type={'Arc'}
        />
      </View>
    );
  }

  return (
    <View style={[styles.spinnerStyle, style]}>
      <ActivityIndicator size={36} color={theme.heal.colors.crimson} />
    </View>
  );
};

const styles = {
  spinnerStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
};

SpinningLoader.propTypes = {
  style: ViewPropTypes.style,
};

SpinningLoader.defaultProps = {
  style: {},
};

export default SpinningLoader;
