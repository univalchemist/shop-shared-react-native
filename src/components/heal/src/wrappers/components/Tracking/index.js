import withTracking from './withTracking';
import {
  Button,
  FloatingActionButton,
  ListItem,
  ListItemPicker,
} from '@cxa-rn/components';

export const TrackedButton = withTracking(Button);
export const TrackedFloatingActionButton = withTracking(FloatingActionButton);
export const TrackedListItem = withTracking(ListItem);
export const TrackedListItemPicker = withTracking(ListItemPicker);

export { withTracking };
export { TrackedCarousel, TrackedCarouselWithScrollBar } from './Carousel';
