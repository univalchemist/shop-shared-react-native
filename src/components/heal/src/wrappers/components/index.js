import React from 'react';

export {
  Image,
  Text,
  PlainText,
  Box,
  Button,
  Divider,
  Flex,
  SectionHeadingText,
  ErrorPanel,
  ListItemWithRightChevron,
  ScrollView,
  Icon,
  Loader,
} from '@cxa-rn/components';
export {
  withTracking,
  TrackedButton,
  TrackedFloatingActionButton,
  TrackedListItem,
  TrackedCarousel,
  TrackedCarouselWithScrollBar,
} from './Tracking';
