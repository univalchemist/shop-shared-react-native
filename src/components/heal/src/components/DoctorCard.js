import React from 'react';
import { TouchableOpacity, StyleSheet, Image } from 'react-native';
import { Box, Text } from '@heal/src/wrappers/components';
import { useIntl, useTheme } from '@heal/src/wrappers/core/hooks';
import { chevronRight, iconTime, doctorBanner } from '@heal/images';
import theme from '@theme';

const styles = StyleSheet.create({
  chevron: { alignSelf: 'center' },
  image: {
    width: 188,
    height: 117,
  },
  card: {
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0.05,
    shadowRadius: 6,
    shadowColor: theme.colors.black,
    backgroundColor: theme.colors.white,
  },
  selectedCard: {
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0.05,
    shadowRadius: 6,
    shadowColor: theme.colors.black,
    backgroundColor: theme.colors.white,
    borderColor: theme.colors.green,
    borderWidth: 1,
  },
});

const DoctorCard = ({
  doctor,
  width,
  height,
  selected,
  icon,
  showIcon,
  onPress,
}) => {
  const theme = useTheme();
  const intl = useIntl();

  return (
    <TouchableOpacity onPress={onPress}>
      <Box
        width={width}
        height={height}
        style={selected ? styles.selectedCard : styles.card}
      >
        <Image
          source={doctorBanner}
          style={styles.image}
          backgroundColor={theme.colors.gray[4]}
        />
        <Box flex={1} mx={16} my={16} flexDirection="row">
          <Box marginRight={16}>
            <Text numberOfLines={2} color="gray.0">
              {doctor.name}
            </Text>
            <Text numberOfLines={2} fontSize={14} lineHeight={20} mt={8}>
              {doctor.specialityCode
                ? intl.formatMessage({
                    id: `speciality.name.${doctor.specialityCode}`,
                  })
                : ''}
            </Text>
            <Box marginTop="auto" flexDirection="row">
              <Image source={iconTime} />
              <Text
                marginLeft={10}
                fontSize={12}
                lineHeight={20}
                color={theme.colors.gray[3]}
              >
                {`${doctor.estimatedConsultationTime} ${intl.formatMessage({
                  id: 'clinic.estimatedTime',
                })}`}
              </Text>
            </Box>
          </Box>
          {showIcon ? (
            <Image
              marginLeft="auto"
              source={icon ? icon : chevronRight}
              style={styles.chevron}
            />
          ) : null}
        </Box>
      </Box>
    </TouchableOpacity>
  );
};

export default DoctorCard;
