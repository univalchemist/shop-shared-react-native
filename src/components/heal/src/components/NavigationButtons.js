import React from 'react';
import { Platform, TouchableOpacity } from 'react-native';
import { Image } from '@heal/src/wrappers/components';
import { navBackArrow } from '@heal/images';

export const StackBackButton = ({ style, onPress }) => {
  return (
    <TouchableOpacity
      activeOpacity={1}
      style={{
        ...Platform.select({
          ios: { paddingLeft: 16 },
          android: {},
        }),
        ...style,
        paddingRight: 16,
      }}
      onPress={onPress}
    >
      <Image source={navBackArrow} width={12} height={22} />
    </TouchableOpacity>
  );
};

export const healNavigationOptions = {
  headerBackImage: StackBackButton,
};
