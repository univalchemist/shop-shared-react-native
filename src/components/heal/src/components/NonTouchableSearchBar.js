import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Box, PlainText } from '@heal/src/wrappers/components';
import { Icon } from 'react-native-elements';
import { useTheme } from '@wrappers/core/hooks';

const NonTouchableSearchBar = ({ placeholder = 'Search doctor', onPress, ...props }) => {
  const theme = useTheme();
  return (
    <TouchableOpacity style={{ flex: 1 }} activeOpacity={0.7} onPress={onPress}>
      <Box
        px={10}
        flexDirection={'row'}
        height={44}
        alignItems={'center'}
        borderRadius={10}
        backgroundColor={theme.heal.colors.gray[1]}
        {...props}
      >
        <Icon size={32} name={'search'} type={'evilicon'} />
        <PlainText ml={8} color={theme.heal.colors.gray[4]}>
          {placeholder}
        </PlainText>
      </Box>
    </TouchableOpacity>
  );
};

export default NonTouchableSearchBar;
