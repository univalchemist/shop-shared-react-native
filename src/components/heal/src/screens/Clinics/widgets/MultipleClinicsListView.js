import React from 'react';
import { FlatList } from 'react-native';
import { Box, Image } from '@cxa-rn/components';
import { ListItemWithRightChevron, Text } from '@wrappers/components';
import { CLINIC_DETAILS } from '@routes';
import { panelDoctor } from '@heal/images';
import { useTheme } from '@wrappers/core/hooks';
import { connect } from 'react-redux';
import { setDetailsClinic } from '@heal/src/store/actions';

const getDoctororDoctors = length => {
  return `${length} ${length > 1 ? 'Doctors' : 'Doctor'}`;
};

const PanelListItem = React.memo(function PanelListItem(props) {
  const { setDetailsClinic, navigation, ...clinic } = props;
  const theme = useTheme();
  return (
    <ListItemWithRightChevron
      accessibilityLabel={clinic.name}
      onPress={() => {
        setDetailsClinic(clinic);
        navigation.navigate(CLINIC_DETAILS, {
          clinic: clinic,
        });
      }}
    >
      <Box>
        <Text color={theme.colors.fonts.blackLink} fontSize={21}>
          {clinic.name}
        </Text>
        <Text fontSize={theme.fontSizes[2]}>{clinic.address}</Text>
        <Box alignItems={'center'} pt={16} flexDirection={'row'}>
          <Image source={panelDoctor} />
          <Text fontSize={theme.fontSizes[1]} marginLeft={8}>
            {getDoctororDoctors(clinic.doctors.length)}
          </Text>
        </Box>
      </Box>
    </ListItemWithRightChevron>
  );
});

const MultipleClinicsListView = ({ navigation, route, setDetailsClinic }) => {
  const clinics = route?.params?.clinics || [];
  const theme = useTheme();
  return (
    <Box backgroundColor={theme.colors.white} flex={1}>
      <FlatList
        data={clinics}
        initialNumToRender={100}
        onEndReachedThreshold={0.5}
        keyExtractor={item => `${item.id}`}
        renderItem={({ item }) => {
          return (
            <PanelListItem
              setDetailsClinic={setDetailsClinic}
              navigation={navigation}
              {...item}
            />
          );
        }}
      />
    </Box>
  );
};

export default connect(null, { setDetailsClinic })(MultipleClinicsListView);
