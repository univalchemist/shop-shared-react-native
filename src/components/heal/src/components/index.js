import { StackBackButton, healNavigationOptions } from './NavigationButtons';
import NonTouchableSearchBar from './NonTouchableSearchBar';
import DoctorCard from './DoctorCard';
export { default as SpinningLoader } from './SpinningLoader';

export {
  DoctorCard,
  StackBackButton,
  healNavigationOptions,
  NonTouchableSearchBar,
};
