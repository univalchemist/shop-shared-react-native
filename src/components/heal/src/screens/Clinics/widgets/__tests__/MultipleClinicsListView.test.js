import React from 'react';
import { renderForTest } from '@testUtils';
import { FlatList } from 'react-native';
import MultipleClinicsListView from '../MultipleClinicsListView';
import { ListItemWithRightChevron } from '@wrappers/components';
import { fireEvent } from 'react-native-testing-library';
import { Image } from '@cxa-rn/components';
describe('MultipleClinicsListView', () => {
  const clinics = [
    {
      address: '2149 JALAN BUKIT RIMAU',
      district: 'some district',
      doctors: [],
      id: 212,
      location: [
        {
          latitude: 22.001232,
          longitude: 114.186422,
          id: 434,
        },
      ],
      name: 'name',
      qrCode: 'somecode',
    },
    {
      address: '2149 JALAN BUKIT RIMAU',
      district: 'some district',
      doctors: [],
      id: 212,
      location: [
        {
          latitude: 22.001232,
          longitude: 114.186422,
          id: 434,
        },
      ],
      name: 'name',
      qrCode: 'somecode',
    },
  ];

  const navigation = {
    navigate: jest.fn(),
  };

  const route = {
    params: {
      clinics,
    },
  };

  it('should render snapshot', () => {
    const component = renderForTest(
      <MultipleClinicsListView navigation={navigation} route={route} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it.only('should render flatlist that renders list of clinics passed', () => {
    const component = renderForTest(
      <MultipleClinicsListView navigation={navigation} route={route} />,
    );

    const flatList = component.queryAllByType(FlatList);

    expect(flatList.length).toBe(1);
    expect(flatList[0].props.data).toEqual(clinics);
  });

  it('should render the same number of list item as the number of clinics', () => {
    const component = renderForTest(
      <MultipleClinicsListView navigation={navigation} route={route} />,
    );

    expect(component.queryAllByType(ListItemWithRightChevron).length).toBe(
      clinics.length,
    );
  });

  it('should render the clinic name and consultation type', () => {
    const component = renderForTest(
      <MultipleClinicsListView navigation={navigation} route={route} />,
    );

    expect(component.queryAllByText(clinics[0].name).length).toBe(1);
    expect(component.queryAllByText(clinics[1].name).length).toBe(1);

    expect(component.queryAllByType(clinics[0].address).length).toBe(1);
    expect(component.queryAllByType(clinics[1].address).length).toBe(1);
    expect(component.getAllByType(Image)).toBeDefined();
  });

  it('should navigate to clinic details on pressed', () => {
    const component = renderForTest(
      <MultipleClinicsListView navigation={navigation} route={route} />,
    );

    const clinicListItem = component.queryAllByType(
      ListItemWithRightChevron,
    )[0];

    fireEvent.press(clinicListItem);

    expect(navigation.navigate).toHaveBeenCalledWith('PanelClinicDetails', {
      selectedClinic: clinics[0],
    });
  });
});
