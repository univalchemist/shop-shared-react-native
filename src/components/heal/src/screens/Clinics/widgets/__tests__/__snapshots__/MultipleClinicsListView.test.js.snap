// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleClinicsListView should render snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTScrollView
                        data={
                          Array [
                            Object {
                              "name": "Dr Abc",
                              "specialty": "General",
                            },
                            Object {
                              "name": "Dr efg",
                              "specialty": "Dentistry",
                            },
                          ]
                        }
                        disableVirtualization={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        horizontal={false}
                        initialNumToRender={100}
                        keyExtractor={[Function]}
                        maxToRenderPerBatch={10}
                        numColumns={1}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={0.5}
                        onLayout={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        updateCellsBatchingPeriod={50}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={21}
                      >
                        <View>
                          <View
                            onLayout={[Function]}
                            style={null}
                          >
                            <View
                              accessibilityLabel="Dr Abc"
                              accessible={true}
                              focusable={true}
                              isTVSelectable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={null}
                            >
                              <View
                                alignItems="center"
                                flexDirection="row"
                                px={4}
                                py={24}
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "paddingBottom": 24,
                                      "paddingLeft": 32,
                                      "paddingRight": 32,
                                      "paddingTop": 24,
                                    },
                                  ]
                                }
                              >
                                <View
                                  flex={1}
                                  style={
                                    Array [
                                      Object {
                                        "flexBasis": 0,
                                        "flexGrow": 1,
                                        "flexShrink": 1,
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={16}
                                    fontWeight="bold"
                                    lineHeight={22}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 16,
                                          "fontWeight": "700",
                                          "lineHeight": 22,
                                        },
                                      ]
                                    }
                                  >
                                    Dr Abc
                                  </Text>
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={16}
                                    lineHeight={22}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 16,
                                          "lineHeight": 22,
                                        },
                                        Object {
                                          "textTransform": "capitalize",
                                        },
                                      ]
                                    }
                                  >
                                    General
                                  </Text>
                                </View>
                                <View
                                  marginLeft={3}
                                  style={
                                    Array [
                                      Object {
                                        "marginLeft": 16,
                                      },
                                    ]
                                  }
                                >
                                  <Image
                                    size={30}
                                    source={
                                      Object {
                                        "testUri": "../../../src/components/commons/ListItemWithRightChevron/images/angleRight.png",
                                      }
                                    }
                                    style={
                                      Array [
                                        Object {
                                          "height": 30,
                                          "width": 30,
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                              <View
                                pl={4}
                                pr={4}
                                style={
                                  Array [
                                    Object {
                                      "paddingLeft": 32,
                                      "paddingRight": 32,
                                    },
                                  ]
                                }
                              >
                                <View
                                  backgroundColor="gray.5"
                                  height={1}
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "#e3e3e3",
                                        "height": 1,
                                        "width": "100%",
                                      },
                                    ]
                                  }
                                  width="100%"
                                />
                              </View>
                            </View>
                          </View>
                          <View
                            onLayout={[Function]}
                            style={null}
                          >
                            <View
                              accessibilityLabel="Dr efg"
                              accessible={true}
                              focusable={true}
                              isTVSelectable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={null}
                            >
                              <View
                                alignItems="center"
                                flexDirection="row"
                                px={4}
                                py={24}
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "paddingBottom": 24,
                                      "paddingLeft": 32,
                                      "paddingRight": 32,
                                      "paddingTop": 24,
                                    },
                                  ]
                                }
                              >
                                <View
                                  flex={1}
                                  style={
                                    Array [
                                      Object {
                                        "flexBasis": 0,
                                        "flexGrow": 1,
                                        "flexShrink": 1,
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={16}
                                    fontWeight="bold"
                                    lineHeight={22}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 16,
                                          "fontWeight": "700",
                                          "lineHeight": 22,
                                        },
                                      ]
                                    }
                                  >
                                    Dr efg
                                  </Text>
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={16}
                                    lineHeight={22}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 16,
                                          "lineHeight": 22,
                                        },
                                        Object {
                                          "textTransform": "capitalize",
                                        },
                                      ]
                                    }
                                  >
                                    Dentistry
                                  </Text>
                                </View>
                                <View
                                  marginLeft={3}
                                  style={
                                    Array [
                                      Object {
                                        "marginLeft": 16,
                                      },
                                    ]
                                  }
                                >
                                  <Image
                                    size={30}
                                    source={
                                      Object {
                                        "testUri": "../../../src/components/commons/ListItemWithRightChevron/images/angleRight.png",
                                      }
                                    }
                                    style={
                                      Array [
                                        Object {
                                          "height": 30,
                                          "width": 30,
                                        },
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                              <View
                                pl={4}
                                pr={4}
                                style={
                                  Array [
                                    Object {
                                      "paddingLeft": 32,
                                      "paddingRight": 32,
                                    },
                                  ]
                                }
                              >
                                <View
                                  backgroundColor="gray.5"
                                  height={1}
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "#e3e3e3",
                                        "height": 1,
                                        "width": "100%",
                                      },
                                    ]
                                  }
                                  width="100%"
                                />
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
